packages/react/document-context/src/document-context.tsx

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+
+// Use null as initial value to handle SSR safely
+const DocumentContext = React.createContext<Document | null>(null);
+
+interface DocumentProviderProps {
+  document?: Document;
+  children: React.ReactNode;
+}
+
+export function DocumentProvider({ document: doc, children }: DocumentProviderProps) {
+  const value = React.useMemo(
+    () => doc ?? (typeof document !== 'undefined' ? globalThis?.document : null),
+    [doc]
+  );
+  return <DocumentContext.Provider value={value}>{children}</DocumentContext.Provider>;
+}
+
+export function useDocument() {
+  const doc = React.useContext(DocumentContext);
+  // Return default document if available and no context value
+  return doc ?? (typeof document !== 'undefined' ? globalThis?.document : null);
+}
```

Comment:
We should check that JS is hydrated before accessing the global document. This prevents potential bugs if referenced during render.
```suggestion
const subscribe = () => () => {};
export function useDocument() {
  const doc = React.useContext(DocumentContext);
  const isHydrated = useSyncExternalStore(
    subscribe,
    () => true,
    () => false
  );
  return doc ?? (isHydrated ? document : null);
}
```

Also want to use the `use-sync-external-store` package since we support older versions of React.
```ts
import { useSyncExternalStore } from 'use-sync-external-store/shim';
```